transcendence-app/src/shared/local-file/local-file.service.ts

```diff
@@ -81,9 +82,13 @@ export class LocalFileService {
       path,
       mimetype,
     );
+    const dataStream = Readable.from(data);
 
-    outStream.write(data);
-    outStream.end();
+    dataStream.pipe(outStream);
+
+    outStream.on('error', (err) => {
```

Comment:
shouldn't we subscribe the listener before we call the pipe on the write stream? Not that I am sure about it...

Reply:
I think you are right. I've updated it. Thanks for the suggestion.

Reply:
I've changed this part to use [`stream.pipeline()`](https://nodejs.org/api/stream.html#streampipelinestreams-callback) which close the streams automatically in case of an error.

From docs:
> stream.pipeline() closes all the streams when an error is raised.